script.reddit.reader/default.py

```diff
@@ -0,0 +1,2105 @@
+                    #xbmcplugin.addDirectoryItem(handle=pluginhandle,url="",listitem=liz,isFolder=False)
+                
+                #END section are for comments that have no links or unsupported links
+    
+    except Exception as e:
+        log('  ' + str(e) )
+        #xbmc.executebuiltin('XBMC.Notification("%s", "%s" )' %( e, 'Flickr' )  )        
+        
+    xbmc_busy(False)
+    #for di in directory_items:
+    #    log( str(di) )
+
+    from resources.lib.guis import commentsGUI
+    
+    li=[]
+    for di in directory_items:
+        #log( '   %s-%s'  %(di[1].getLabel(), di[1].getProperty('onClick_action') ) )
+        li.append( di[1] )
+    
+#     li.sort( key=getKey )
+#     log("  sorted")
+#     
+#     for l in li:
+#         log( '   %s-%s'  %(l.getLabel(), l.getProperty('onClick_action') ) )
+            
+        
+    ui = commentsGUI('view_461_comments.xml' , addon_path, defaultSkin='Default', defaultRes='1080i', listing=li, id=55)
+    #NOTE: the subreddit selection screen and comments screen use the same gui. there is a button that is only for the comments screen
+    ui.setProperty('comments', 'yes')   #i cannot get the links button to show/hide in the gui class. I resort to setting a property and having the button xml check for this property to show/hide
+    
+    #ui = commentsGUI('view_463_comments.xml' , addon_path, defaultSkin='Default', defaultRes='1080i', listing=li, id=55)
+    ui.title_bar_text=post_title
+    ui.include_parent_directory_entry=False
+
+    ui.doModal()
+    del ui
+
+harvest=[]
+def r_linkHunter(json_node,d=0):
+    from resources.lib.domains import url_is_supported
+    from resources.lib.utils import unescape
+    #recursive function to harvest stuff from the reddit comments json reply
+    prog = re.compile('<a href=[\'"]?([^\'" >]+)[\'"]>(.*?)</a>')   
+    for e in json_node:
+        link_desc=""
+        link_http=""
+        author=""
+        created_utc=""
+        if e['kind']=='t1':     #'t1' for comments   'more' for more comments (not supported)
+        
+            #log("replyid:"+str(d)+" "+e['data']['id'])
+            body=e['data']['body'].encode('utf-8')
+    
+            #log("reply:"+str(d)+" "+body.replace('\n','')[0:80])
+            
+            try: replies=e['data']['replies']['data']['children']
+            except: replies=""
+            
+            try: score=e['data']['score']
+            except: score=0
+            
+            try: post_text=unescape( e['data']['body'].encode('utf-8') )
+            except: post_text=""
+            post_text=post_text.replace("\n\n","\n")
+            
+            try: post_html=unescape( e['data']['body_html'].encode('utf-8') )
+            except: post_html=""
+    
+            try: created_utc=e['data']['created_utc']
+            except: created_utc=""
+    
+            try: author=e['data']['author'].encode('utf-8')
+            except: author=""
+    
+            
```

Comment:
I assume that's a guide for you.
Mind removing that? (I mean the whole thing aaaaaaall the way down)